include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -5367,6 +5384,521 @@ struct OrtApi {
+   * The OrtArrayOfConstObjects instance does not own the underlying objects, only the pointers
+   * to them.
+   *
+   * An OrtArrayOfConstObjects instance stores elements of type 'const void*'. Users
+   * must check the object's type via ArrayOfConstObjects_GetObjectType before casting objects
+   * to their actual type.
+   *
+   * \param[in] object_type The object's type as indicated by the OrtTypeTag enum.
+   * \param[in] initial_size The backing array's initial size. Can be set to 0.
+   * \param[in] initial_value Each element's initial value. Can be set to NULL.
+   * \param[out] out A pointer to a newly created OrtArrayOfConstObjects instance.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \note Must be released by calling ReleaseArrayOfConstObjects.
+   *
+   * \since Version 1.23.
+   */
+  ORT_API2_STATUS(CreateArrayOfConstObjects, _In_ OrtTypeTag object_type, _In_ size_t initial_size,
+                  _In_ const void* initial_value, _Outptr_ OrtArrayOfConstObjects** out);
+
+  ORT_CLASS_RELEASE(ArrayOfConstObjects);
+
+  /** \brief Get a tag that represents the type of the opaque objects stored in a OrtArrayOfConstObjects instance.
+   *
+   * Refer to OrtTypeTag for valid values.
+   *
+   * \param[in] array The OrtArrayOfConstObjects instance.
+   * \param[out] type_tag Output parameter set to the type tag that corresponds to the object type.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.23.
+   */
+  ORT_API2_STATUS(ArrayOfConstObjects_GetObjectType, _In_ const OrtArrayOfConstObjects* array,
+                  _Out_ OrtTypeTag* type_tag);
+
+  /** \brief Get a pointer to a data buffer of contiguous elements, where each element is a pointer to a
+   * constant opaque object (i.e., each element is a 'const void*').
+   *
+   * Caller must cast the objects to the appropriate type indicated by ArrayOfConstObjects_GetObjectType.
+   *
+   * \param[in] array The OrtArrayOfConstObjects instance.
+   * \param[out] data Output parameter set to the contiguous data buffer that stores all elements.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.23.
+   */
+  ORT_API2_STATUS(ArrayOfConstObjects_GetData, _In_ OrtArrayOfConstObjects* array, _Outptr_ const void*** data);
+
+  /** \brief Get a pointer to a data buffer of contiguous elements, where each element is a constant pointer to a
+   * constant opaque object (i.e., each element is a 'const void* const').
+   *
+   * Caller must cast the objects to the appropriate type indicated by ArrayOfConstObjects_GetObjectType.
+   *
+   * \param[in] array The OrtArrayOfConstObjects instance.
+   * \param[out] data Output parameter set to the contiguous data buffer that stores all elements.
+   *
+   * \snippet{doc} snippets.dox OrtStatus Return Value
+   *
+   * \since Version 1.23.
+   */
+  ORT_API2_STATUS(ArrayOfConstObjects_GetConstData, _In_ const OrtArrayOfConstObjects* array,
```

Comment:
maybe call them GetData (const) and GetMutableData (not const) for consistency, e.g., with GetTensorMutableData

Reply:
Updated. Renamed to be more consistent.